src/lib/Component.coffee

```diff
@@ -408,12 +419,18 @@ class ProcessInput
     args = ['in'] unless args.length
     if typeof args[args.length - 1] is 'function'
       validate = args.pop()
-      for port in args
-        return false unless @ports[port].has @scope, validate
-      return true
-    res = true
-    res and= @ports[port].ready @scope for port in args
-    res
+    else
+      validate = -> true
+    for port in args
+      if Array.isArray port
+        unless @ports[port[0]].isAddressable()
+          throw new Error "Non-addressable ports, access must be with string #{port[0]}"
+        return false unless @ports[port[0]].has @scope, port[1], validate
+        continue
+      if @ports[port].isAddressable()
+        throw new Error "For addressable ports, access must be with array [#{port}, idx]"
```

Comment:
Yeah, though if you don't care about index then better not to use an addressable port. But indeed, `input.has('portname')` could probably wrap attached and the loop for convenience